pos_payment_method_cashdro/static/src/js/screens.js

```diff
@@ -0,0 +1,243 @@
+            url += "&operation=askOperation";
+            url += "&operationId=" + operation_id;
+            return url;
+        },
+
+        _cashdro_finish_url: function(operation_id) {
+            // Compose the url for a sale report to Cashdro
+            var url = this._cashdro_url()
+            url += "&operation=finishOperation&type=2";
+            url += "&operationId=" + operation_id;
+            return url;
+        },
+
+        _cashdro_request: function (url) {
+            // We'll use it for regular requests
+            return $.ajax({
+                url: url,
+                method: "GET",
+                async: true,
+                success: function (response) {
+                    return response;
+                }
+            });
+        },
+
+        _cashdro_request_payment: function (url) {
+            // This is a special request, as we keep requesting the CashDro
+            // until we get the *finished* state that will give us the amount
+            // received in the cashdrawer.
+            var def = $.Deferred();
+            var _request_payment = function(url) {
+              $.ajax({
+                  url: url,
+                  method: "GET",
+                  success: function (response) {
+                      var data = JSON.parse(response.data);
+                      if (data.operation.state === "F") {
+                          def.resolve(response);
+                      } else {
+                          _request_payment(url);
+                      }
+                  }
+              });
+            }
+            _request_payment(url);
+            return def;
+        },
+
+
+        _show_error: function (msg, title) {
+            if (!title) {
+                title =  _t("CashDro Error");
+            }
+            this.pos.gui.show_popup("error", {
+                "title": title,
+                "body": msg,
+            });
+        },
+
+        // These methods should be covered in v13 with the new payment methods
+
+        _get_journal: function (line_cid) {
+            var line;
+            var order = this.pos.get_order();
+            var lines = order.get_paymentlines();
+            for ( var i = 0; i < lines.length; i++ ) {
+                if (lines[i].cid === line_cid) {
+                    line = lines[i];
+                }
+            }
+            return line.cashregister.journal
+        },
+        _cancel_current_transaction_popup: function () {
+            var self = this;
+            this.gui.show_popup("confirm",{
+                title: _t("Transaction ongoing"),
+                body:  _t("There is a transaction in progress, would you like to cancel it?"),
+                confirm: function () {
+                    self._cancel_current_transaction()
+                },
+                cancel: function () {
+                    return;
+                }
+            });
+        },
+
+    });
+
+});
```

Comment:
```suggestion
    return screens;
});
```

This helps other modules potentially extend this one